src/seedu/addressbook/data/person/Address.java

```diff
@@ -23,10 +28,39 @@
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
+        this.value = trimmedAddress;
+        splitAddress(trimmedAddress);
+        return;
+
+    }
+
+    private void splitAddress(String trimmedAddress) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        } else {
```

Comment:
If it reaches this branch is it guarantee to be a valid address? Are there any address string that would cause incomplete instantiation?